### ISSUE TITLE
missing _class causes failure when deserializing to ObjectRecord

### ISSUE THREAD

**Issue by lrozek** [spring-projects/spring-data-redis#2251]
`StreamMessageListenerContainerOptions#targetType`
https://github.com/lrozek/redis-stream-deserialization/blob/main/redis-stream-consumer/src/main/java/pl/lrozek/redis/stream/consumer/redis/configuration/RedisListenerConfiguration.java#L29
 provides information regarding what class should be used for deserialization. Producing system should not be enforced to provide `_class pl.lrozek.redis.stream.consumer.domain.TemperatureReadingDto` field-value pair in a stream entry, as this is an implementation detail of a consuming system.

When `_class` field-value pair is missing following exception is thrown:
```
2022-01-28 13:17:51.472 ERROR 1 --- [cTaskExecutor-1] ageListenerContainer$LoggingErrorHandler : Unexpected error occurred in scheduled task.

org.springframework.core.convert.ConversionFailedException: Failed to convert from type [org.springframework.data.redis.connection.stream.StreamRecords$ByteMapBackedRecord] to type [pl.lrozek.redis.stream.consumer.domain.TemperatureReadingDto] for value 'MapBackedRecord{recordId=1643375871467-0, kvMap={[B@1f6764b8=[B@2da89d71, [B@53d5009=[B@666e28c7}}'; nested exception is java.lang.IllegalArgumentException: Value must not be null!
	at org.springframework.data.redis.stream.StreamPollTask.convertRecord(StreamPollTask.java:198) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.stream.StreamPollTask.deserializeAndEmitRecords(StreamPollTask.java:176) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.stream.StreamPollTask.doLoop(StreamPollTask.java:148) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.stream.StreamPollTask.run(StreamPollTask.java:132) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: java.lang.IllegalArgumentException: Value must not be null!
	at org.springframework.util.Assert.notNull(Assert.java:201) ~[spring-core-5.3.15.jar:5.3.15]
	at org.springframework.data.redis.connection.stream.Record.of(Record.java:81) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.connection.stream.MapRecord.toObjectRecord(MapRecord.java:147) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.core.StreamObjectMapper.toObjectRecord(StreamObjectMapper.java:138) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.core.StreamOperations.map(StreamOperations.java:577) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.stream.DefaultStreamMessageListenerContainer.lambda$getDeserializer$2(DefaultStreamMessageListenerContainer.java:240) ~[spring-data-redis-2.6.1.jar:2.6.1]
	at org.springframework.data.redis.stream.StreamPollTask.convertRecord(StreamPollTask.java:196) ~[spring-data-redis-2.6.1.jar:2.6.1]
	... 4 common frames omitted

```

reproduction project and full description can be found at
https://github.com/lrozek/redis-stream-deserialization

**Comment by lrozek** [comment]
https://github.com/spring-projects/spring-data-redis/blob/main/src/main/java/org/springframework/data/redis/hash/ObjectHashMapper.java#L177

when `converter.read` is invoked, value for the first param could be provided as `StreamMessageListenerContainerOptions#targetType`  instead of `Object.class`

**Comment by christophstrobl** [comment]
Thanks for reporting! The `HashMapper` currently does not have means to accept a type argument directly. 
We'll look into this matter.

**Comment by mp911de** [comment]
Closing as duplicate of #2198

**Comment by lrozek** [comment]
@christophstrobl I noticed that https://github.com/spring-projects/spring-data-redis/pull/2253 was merged to main branch and should also fix this issue. When `spring-data-redis:2.7.0-SNAPSHOT` will be built with your merge to test it ? Any other way to test it before it ?

**Comment by christophstrobl** [comment]
@lrozek SNAPSHOT builds are available via _repo.spring.io_, `/libs-snapshot`.

**Comment by lrozek** [comment]
@christophstrobl I did use `repo.spring.io, /libs-snapshot` but snapshot was not built until `10-02-22 11:44:26` that's why I was getting older built.

Now with the latest version I can also confirm that https://github.com/spring-projects/spring-data-redis/pull/2253 resolved also this issue. I verified it in repro project with following change to `2.7.0-SNAPSHOT` at https://github.com/lrozek/redis-stream-deserialization/commit/fc174cc3dc9f139783d50fcc109db2f7cafcad40

What's it the schedule to have `2.7.0` version out?

Thank you
